File: utils/res_analyzer.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Record {
public:
    int ae, acc;
    double re;
    Record(const int real, const int my) {
        ae = real - my;
        ae = ae < 0 ? -ae : ae;
        re = ae / (double)real;
        acc = real == my;
    }
};

int main() {
    int n;
    cin >> n;
    vector<string> name(n, "");
    for (int i = 0; i < n; ++i) {
        cin >> name[i];
    }
    vector<vector<Record> > rec;
    int real, my;
    while (cin >> real) {
        rec.push_back(vector<Record>());
        for (int i = 0; i < n; ++i) {
            cin >> my;
            rec.back().push_back(Record(real, my));
        }
    }
    for (int i = 0; i < n; ++i) {
        cout << name[i] << ":" << endl;
        int ae = 0, acc = 0;
        double re = 0;
        int cdf[10][2];
        for (int j = 0; j < 20; ++j) {
            cdf[j/2][j%2] = 0;
        }
        for (const auto& recc: rec) {
            ae += recc[i].ae;
            acc += recc[i].acc;
            re += recc[i].re;
            for (int j = 0; j < 10; ++j) {
                if (recc[i].ae <= j) ++cdf[j][0];
                if (recc[i].re <= (double)j / 10) ++cdf[j][1];
            }
        }
        cout << "aae: " << (double)ae / rec.size() << endl;
        cout << "are: " << (double)re / rec.size() << endl;
        cout << "acc: " << (double)acc / rec.size() << endl;
        cout << "ae-cdf:";
        for (int j = 0; j < 10; ++j) cout << " " << (double)cdf[j][0] / rec.size();
        cout << endl;
        cout << "re-cdf:";
        for (int j = 0; j < 10; ++j) cout << " " << (double)cdf[j][1] / rec.size();
        cout << endl;
    }
}

File: utils/res_analyzer.py
import sys
def read_file(name):
    with open(name) as f:
        res = []
        for line in f:
            res.append(line.split())
        names = res[0][1:]
        data = []
        for line in res[1:]:
            data.append([int(n) for n in line])
        return names, data
    return []

def cdf(arr, thres):
    res = [0 for thr in thres]
    for n in arr:
        for i in range(len(thres)):
            if n <= thres[i]:
                res[i] += 1

    return [r / len(arr) for r in res]

def analyze(r, e):
    ae = [abs(r[i] - e[i]) for i in range(len(r))]
    re = [ae[i] / r[i] for i in range(len(r))]
    aae = (sum(ae) / len(ae))
    are = (sum(re) / len(re))
    acc = (len([aei for aei in ae if aei == 0]) / len(ae))
    ae_cdf = cdf(ae, list(range(10)))
    re_cdf = cdf(re, [n / 10 for n in range(10)])
    return {
        'aae': aae,
        'are': are,
        'acc': acc,
        'ae_cdf': ae_cdf,
        're_cdf': re_cdf,
    }


names, data = read_file(sys.argv[1])
reses = []
for i in range(1, len(names) + 1):
    res = analyze([dat[0] for dat in data], [dat[i] for dat in data])
    res['name'] = names[i-1]
    reses.append(res)

import pickle
with open(sys.argv[2], "wb") as f:
    pickle.dump(reses, f)